cicd/forgeops-tests/tests/smoke/ds/test_ds.py

```diff
@@ -56,32 +60,39 @@ def test_resource_replication(self):
             "_schema": "frapi:opendj:rest2ldap:user:1.0"
         }
 
-        logging.info("Creating a new user entry with ID newuser_ds0")
+        logger.test_step('Creating a new user entry with ID newuser_ds0')
         response = put(verify=self.dscfg.ssl_verify, url=self.dscfg.ds0_url + '/api/users/newuser_ds0',
-                   auth=('am-identity-bind-account', 'password'), headers=headers,json=json_data)
-        self.assertEqual(response.status_code, 201)
+                       auth=('am-identity-bind-account', 'password'), headers=headers, json=json_data)
+        rest.check_http_status(http_result=response, expected_status=201)
 
-        logging.info("Verifying user entry with ID newuser_ds0 replicated in userstore-1")
+        logger.test_step('Verifying user entry with ID newuser_ds0 replicated in userstore-1')
         response = get(verify=self.dscfg.ssl_verify, url=self.dscfg.ds1_url + '/api/users/newuser_ds0',
                        auth=('am-identity-bind-account', 'password'))
-        self.assertEqual(response.status_code, 200)
+        rest.check_http_status(http_result=response, expected_status=200)
 
-    @classmethod
-    def tearDownClass(self):
-        instance = DSSmoke()
-        """Delete datastore resource from users/"""
+    @pytest.fixture()
+    def setup_teardown_test_resource_replication(self):
+        """"Setup and Teardown for test_resource_replication"""
 
-        logging.info("Deleting user entry with ID newuser_ds0 from userstore-0")
+        yield "resource"
```

Comment:
I haven't grasped fixtures yet.  But does "resource" map to test_1_resource_replication?

Reply:
Yes in fact it seems that you can put what you want in the string. 
I used "resource" because I found that in an example but you can use what you want, I don't where this info/string appear.
Looks like the only important stuff is the `yield` call which calls the calling method.